### PR TITLE
Overview mixer card

### PR DESCRIPTION
**Pull request by crux25**
closes https://github.com/crypto-power/cryptopower/issues/83

This MR adds live data to the mixer card on the overview page.

### CODE REVIEW

**Review by dreacot — CHANGES_REQUESTED** [review]
per the figma design, this is meant to take full width, when mixer isn't running

Screenshot from 2023-09-27 18-12-27

the right side of the mixer card has inconsistent padding, the gray area overflows

Screenshot from 2023-09-27 18-20-18

**Review by ukane-philemon — CHANGES_REQUESTED** [review]
Screenshot 2023-09-28 at 2 33 39 PM

The wallet card does not align with the design(smaller and still takes full width when there's no mixing card) and the top left and bottom left border needs a little attention.

PS: I need help with finding the option to toggle mixing so I can see the view too.

**Comment by dreacot** [comment]
>PS: I need help with finding the option to toggle mixing so I can see the view too.

@ukane-philemon open a decred wallet, click o stakeshuffle, you can start mixing there(ensure you have funds in your unmixed accoun)

Screenshot from 2023-09-28 07-53-27

File: ui/page/root/overview_page.go [hunk 1]
```diff
@@ -666,3 +722,126 @@ func (pg *OverviewPage) centerLayout(gtx C, top, bottom unit.Dp, content layout.
...
+	for _, wal := range dcrWallets {
+		w := wal.(*dcr.Asset)
+
+		if w.IsAccountMixerActive() {
+			if _, ok := pg.mixerSliderData[w.ID]; !ok {
+				pg.mixerSliderData[w.ID] = &mixerData{
+					Asset: w,
+				}
+				pg.setUnMixedBalance(w.ID)
+				// Store the slide keys in a slice to maitain a consistence slide sequence.
```
Comment (ukane-philemon) [inline ui/page/root/overview_page.go:765]: nit: maitain

**Review by ukane-philemon — CHANGES_REQUESTED** [review]

**Review by dreacot — CHANGES_REQUESTED** [review]
Stopping the mixer doesn't update the overview page

step 1: i click the arrow to take me to the mixer page

Screenshot from 2023-09-28 08-05-09

step 2: i cancel the mixer operation for my wallet (qwerty)

Screenshot from 2023-09-28 08-06-05

step 3: i head back to the overview page, but the mixer card for my wallet (qwerty) is still showing

Screenshot from 2023-09-28 08-06-33

**Comment by ukane-philemon** [comment]
RE: https://github.com/crypto-power/cryptopower/pull/118#pullrequestreview-1648898561

I've seen the updated designs. Only the borders need a little touchup.

**Comment by Sirmorrison** [comment]
I just noticed a regression somewhere in the image layoutSize2() code base that should fix the layout issue @crux25 is having.

Should be  `dst := image.NewRGBA(image.Rectangle{Max: image.Point{X: int(width), Y: int(height * 2)}})`

not `dst := image.NewRGBA(image.Rectangle{Max: image.Point{X: int(width * 2), Y: int(height * 2)}})`

the regression was introduced in this PR https://github.com/crypto-power/cryptopower/pull/95, however from the issue opened i see it fixed a layout bug on linux.

so i will recommend leaving the card layout as it is as it is not from this PR and we can have and issue that will resolve the layout bug.


> Screenshot 2023-09-28 at 2 33 39 PM
> The wallet card does not align with the design(smaller and still takes full width when there's no mixing card) and the top left and bottom left border needs a little attention.
> 
> PS: I need help with finding the option to toggle mixing so I can see the view too.

**Comment by crux25** [comment]
> Stopping the mixer doesn't update the overview page
> 
> step 1: i click the arrow to take me to the mixer page
> 
Resolved.

**Comment by dreacot** [comment]
> so i will recommend leaving the card layout as it is as it is not from this PR and we can have and issue that will resolve the layout bug.

please link an issue when it has been opened

File: ui/page/root/overview_page.go [hunk 2]
```diff
@@ -291,13 +333,24 @@ func (pg *OverviewPage) assetBalanceItemLayout(item assetBalanceSliderItem) layo
 }
 
 func (pg *OverviewPage) mixerSliderLayout(gtx C) D {
-	sliderWidget := []layout.Widget{
-		pg.mixerLayout,
+	sliderWidget := make([]layout.Widget, 0)
+	for _, key := range pg.sortedMixerSlideKeys {
+		// Append the mixer slide widgets in an ananymouse function. This stops the
```
Comment (dreacot) [inline ui/page/root/overview_page.go:338]:
```suggestion
		// Append the mixer slide widgets in an anonymous function. This stops the
```

File: ui/page/root/overview_page.go [hunk 3]
```diff
@@ -766,3 +823,133 @@ func (pg *OverviewPage) centerLayout(gtx C, top, bottom unit.Dp, content layout.
...
+	for _, wal := range dcrWallets {
+		w := wal.(*dcr.Asset)
+
+		if w.IsAccountMixerActive() {
+			if _, ok := pg.mixerSliderData[w.ID]; !ok {
+				pg.mixerSliderData[w.ID] = &mixerData{
+					Asset: w,
+				}
+				pg.setUnMixedBalance(w.ID)
+				// Store the slide keys in a slice to maintain a consistence slide sequence.
```
Comment (dreacot) [inline ui/page/root/overview_page.go:866]:
```suggestion
				// Store the slide keys in a slice to maintain a consistent slide sequence.
```

**Review by dreacot — CHANGES_REQUESTED** [review]

**Comment by dreacot** [comment]
> Stopping the mixer doesn't update the overview page
> 
> step 1: i click the arrow to take me to the mixer page
> 
> Screenshot from 2023-09-28 08-05-09
> 
> step 2: i cancel the mixer operation for my wallet (qwerty)
> 
> Screenshot from 2023-09-28 08-06-05
> 
> step 3: i head back to the overview page, but the mixer card for my wallet (qwerty) is still showing
> 
> Screenshot from 2023-09-28 08-06-33

i repeated this step and got a crash, see the log below

```
panic: assignment to entry in nil map

goroutine 167 [running]:
github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).listenForMixerNotifications(0xc003e35b00)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/overview_page.go:862 +0x4db
github.com/crypto-power/cryptopower/ui/page/root.(*OverviewPage).OnNavigatedTo(0xc003e35b00)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/overview_page.go:161 +0x168
github.com/crypto-power/cryptopower/app.(*PageStack).Pop(0xc002a8d710)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/pagestack.go:93 +0x138
github.com/crypto-power/cryptopower/app.(*MasterPage).CloseCurrentPage(0xc0005de370)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/app/masterpage.go:56 +0x25
github.com/crypto-power/cryptopower/ui/page/root.(*MainPage).HandleUserInteractions(0xc01be27500)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/main_page.go:416 +0x163
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).HandleUserInteractions(0xc004027b00)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/page/root/home_page.go:131 +0x67
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0xc000191c20, {{0xc13ddb7f3a26a185, 0xe5055fbdd, 0x2e53a40}, {0x3f800000, 0x3f800000}, {0x320, 0x28a}, {0x0, 0x0, ...}, ...})
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/window.go:233 +0x14c
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0xc000191c20)
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/ui/window.go:206 +0x2b9
main.main.func3()
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:110 +0x1d
created by main.main
        /home/kennedy/Projects/src/github.com/cryptopower/cryptopower/main.go:108 +0x61b
exit status 2
```

**Review by dreacot — APPROVED** [review]

**Comment by Sirmorrison** [comment]
Associated issue https://github.com/crypto-power/cryptopower/issues/140

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**